core/src/test/java/org/infinispan/executors/LimitedExecutorTest.java

```diff
@@ -57,11 +57,13 @@ public void testConcurrencyLimit() throws Exception {
       limitedExecutor.execute(() -> cf2.complete("bla"));
 
       Thread.sleep(10);
+      assertFalse(cf1.isDone());
       assertFalse(cf2.isDone());
 
       latch.countDown();
       assertEquals("bla", cf1.get(10, SECONDS));
       assertEquals("bla", cf2.get(10, SECONDS));
+      eventuallyEquals(0, executor::getActiveCount);
```

Comment:
Yes, I realized I actually needed 2 threads for `testConcurrencyLimit()`.
But I added another `eventuallyEquals()` call at the beginning of the test methods that use the executor.